File: prototype/Assets/microcosmicWar/Scripts/item/IitemObject.js


virtual function canUse(pGameObject:GameObject):boolean
{
}

virtual function use()
{
}

function Reset()
{
	var lItemObjectImp:ItemObjectImp = zzUtilities.needComponent(gameObject,ItemObjectImp);
	lItemObjectImp.setItemObject(this);
}

File: prototype/Assets/microcosmicWar/Scripts/item/IitemObject.cs

using UnityEngine;
using System.Collections;

public abstract class IitemObject : MonoBehaviour
{


    public abstract bool canUse(GameObject pGameObject);

    public abstract void use();

    void Reset()
    {
        ItemObjectImp lItemObjectImp = (ItemObjectImp)zzUtilities.needComponent(gameObject, typeof(ItemObjectImp));
        lItemObjectImp.setItemObject(this);
    }
}